### ISSUE TITLE
Mapping Is Very Slow From Large Dynamics

### ISSUE THREAD

**Issue by maembe** [SlapperAutoMapper/Slapper.AutoMapper#62]
I have a query that returns an IEnumerable of nested objects.  The query returns about 120,000 rows and 22 columns.  When I log the timings the query (using Dapper), returns the dynamic in about 3 seconds.  Then it takes about 65 seconds to map to my objects (about 500 after the mapping from the 120k rows).   
 The use case is relatively simple:

```
// dapper query to get the dynamic, takes 3 seconds
dynamic res = conn.Query<dynamic>(sql, param:parameters, commandTimeout: timeout);

// map with automapper, takes 65 seconds
var myObjects = Slapper.AutoMapper.MapDynamic<MyObject>(res.Result) as IEnumerable<MyObject>;
```

It's a relatively large set of data, but it doesn't seem large enough to be this slow.  Has any else encountered a similar issue?

EDIT:  I decided to do the mapping by hand from the dynamic for comparison using linq.  It took less that 1 second to run.

**Comment by Dolfik1** [comment]
I has encountered a similar issue. I writing Slapper compatible library - [Mappy](https://github.com/Dolfik1/Mappy). The work in progress right now, but most of Slapper features are supported. This library is 3.5x faster than Slapper on simple objects, and 7x faster on complex objects, and I still trying to improve performance. Also Mappy has less exponential performance loss than Slapper, because does not use reflection at runtime (actually does, but only one time on first call).
Can you measure Mappy performance in your case? The library available in [NuGet](https://www.nuget.org/packages/Mappy/1.0.0-preview-1).

**Comment by randyburden** [comment]
Hi, we've just released version 2.0.3 release which has some performance improvements yielding a 57% mapping speed improvement.
   - v2.0.2: Mapped 120,000 deeply nested complex objects in 30333 ms.
   - v2.0.3: Mapped 120,000 deeply nested complex objects in 13046 ms.

Hopefully, this will help folks out and without any loss of backward compatibility they've come to enjoy with Slapper.AutoMapper.

Reflection is inherently slow, especially using the straight-forward pattern that Slapper.AutoMapper uses. If we were to use advanced techniques such as compiling lambdas or emitting IL I'm sure we could massively improve the performance but it would require a total rewrite, which is what Mappy seems to be.
